File: 3142-check-if-grid-satisfies-conditions/3142-check-if-grid-satisfies-conditions.py
class Solution:
    def satisfiesConditions(self, grid: List[List[int]]) -> bool:
        rows = len(grid)
        cols = len(grid[0])
        
        def checker(i, j):
            nonlocal rows, cols
            if i < rows -1 and grid[i][j] != grid[i+1][j]:
                return False
            if j < cols - 1 and grid[i][j] == grid[i][j+1]:
                return False
            return True
        
        for i in range(rows):
            for j in range(cols):
                if not checker(i, j):
                    return False
        return True

File: 3142-check-if-grid-satisfies-conditions/3142-check-if-grid-satisfies-conditions.java
class Solution {
    public boolean satisfiesConditions(int[][] grid) {
        int rows = grid.length;
        int cols = grid[0].length;
        
        for (int i=0; i<rows; i++){
            for (int j=0; j<cols; j++){
                if (i < rows - 1 && grid[i][j] != grid[i+1][j]){
                    return false;
                }
                if (j < cols - 1 && grid[i][j] == grid[i][j+1]){
                    return false;
                }
            }
        }
        return true;
    }
}